File: src/deck.js
Deck = function () {
    var playingCards = [];

    _.each(PlayingCard.suits, function (suite) {
        _.each(PlayingCard.ranks, function (rank) {
            playingCards.push(new PlayingCard(rank, suite))
        });
    });

    this.shuffle = function () {
        playingCards = _.shuffle(playingCards);
    };

    this.playingCards = function () {
        return playingCards;
    };

    this.getPlayingCard = function () {
        var card, index;
        card = _.sample(playingCards);
        index = playingCards.indexOf(card);
        playingCards.splice(index, 1);
        return card;
    }
};

File: ruby/lib/deck.rb
class Deck < Object

  attr_accessor :rank, :suit

  def initialize
    @playing_cards = []
    Card.suit.values.each { |suit|
      Card.rank.values.each { |rank|
        @playing_cards << Card.new({:rank => rank, :suit => suit})
      }
    }
    self.shuffle
  end

  def playing_cards
    @playing_cards
  end

  def get_playing_card
    card = @playing_cards.sample
    @playing_cards.delete(card)
    card
  end

  def shuffle
    @playing_cards.shuffle
  end
end